### PR TITLE
[test] Increase Browserstack worker timeout from 2.5 to 4 minutes

### PR DESCRIPTION
**Pull request by flaviendelangle**
It doesn't seem to be enough

This PR workflows :

- https://app.circleci.com/pipelines/github/mui-org/material-ui-x/5774/workflows/7d39c3e8-0ac0-4b28-b1b5-e7f4e5879518/jobs/29973

#2008 workflows (old timeout) :

- https://app.circleci.com/pipelines/github/mui-org/material-ui-x/5772/workflows/8554a64a-e189-4061-8d37-1caf63b45f6c
- https://app.circleci.com/pipelines/github/mui-org/material-ui-x/5772/workflows/ad3a635c-5ebb-4e12-ad1b-1aef6d2f5a71
- https://app.circleci.com/pipelines/github/mui-org/material-ui-x/5772/workflows/2dda8897-9da8-4ee7-8a37-ed08024f12c6

#2034 workflows (old timeout) :

- https://app.circleci.com/pipelines/github/mui-org/material-ui-x/5753/workflows/f530e51a-5dfb-4460-9005-4e5d232d3c8f/jobs/29824
- https://app.circleci.com/pipelines/github/mui-org/material-ui-x/5753/workflows/7970ee1b-2e0c-47d4-a4fd-cd9bc0f1f9fb/jobs/29939
- https://app.circleci.com/pipelines/github/mui-org/material-ui-x/5753/workflows/5d2a373a-8e0a-4d39-b144-cd9dcbbb7e90/jobs/29955
- https://app.circleci.com/pipelines/github/mui-org/material-ui-x/5753/workflows/fd09ae4f-0c03-47fe-a205-747d9d7d160d/jobs/29965

Each time it's a timeout

### CODE REVIEW

File: test/karma.conf.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ const browserStack = {
   accessKey: process.env.BROWSERSTACK_ACCESS_KEY,
   build,
   // https://github.com/browserstack/api#timeout300
-  timeout: 2.5 * 60, // Maximum time before a worker is terminated. Default 5 minutes.
+  timeout: 5 * 60, // Maximum time before a worker is terminated. Default 5 minutes.
```
Comment (oliviertassinari) [inline test/karma.conf.js:17]: For context, the higher the value is, the more we potentially slow down the CI when CircleCI timeout (he can no longer cancel the build on BrowserStack.
```suggestion
  timeout: 5 * 60, // Maximum time before a worker is terminated. Default 5 minutes.
```

**Review by oliviertassinari — COMMENTED** [review]
It seems fair

File: test/karma.conf.js (hunk 1 above)
Comment (flaviendelangle) [reply test/karma.conf.js:17]: But it still getting to timeout :/

Comment (oliviertassinari) [reply test/karma.conf.js:17]: Screenshot_2021-07-05-17-11-54-878_com brave browser

Maybe, but the increase of timeout worked. Waited 5 minutes, but not BrowserStack runner was available, before CircleCI timeouted.

Comment (flaviendelangle) [reply test/karma.conf.js:17]: It passed the 2nd time : v

Comment (oliviertassinari) [reply test/karma.conf.js:17]: And it wasn't this slow
Screenshot_2021-07-05-17-36-10-111_com brave browser

Comment (flaviendelangle) [reply test/karma.conf.js:17]: Maybe we can try 3mn30 or 4mn to avoid slowing the CI to much

Comment (oliviertassinari) [reply test/karma.conf.js:17]: 4 minutes sounds great. Plus, it won't matter so much once we stop our current "fire & forget" strategy, triggering builds in BrowserStack that we know will complete after the 30minutes timeout delay of CircleCI for builds.

Comment (flaviendelangle) [reply test/karma.conf.js:17]: If we have a way to fail the step before the 30mn timeout, that would be a great improvement.
I'm setting it to 4mn then.

Comment (oliviertassinari) [reply test/karma.conf.js]: Yeah, needs to rewrite the Karma plugin for this, it's not really hard, but could take a day, we avoided it so far 😁. Or we kill the browser stack tests, or we pay, or we migrate to a Docker selenium grid.

Comment (flaviendelangle) [reply test/karma.conf.js]: OK, I suppose it's worth seeing how it behaves with this PR

**Comment by oliviertassinari** [comment]
I'm merging a bit ahead of phase, as our test keeps failing.

**Pull request merged**